internal/apischema/openai/openai.go

```diff
@@ -149,6 +149,28 @@ func (c *ChatCompletionContentPartUserUnionParam) UnmarshalJSON(data []byte) err
 	return nil
 }
 
+type StringOrAssistantRoleContentUnion struct {
+	Value interface{}
+}
+
+func (s *StringOrAssistantRoleContentUnion) UnmarshalJSON(data []byte) error {
+	var str string
+	err := json.Unmarshal(data, &str)
+	if err == nil {
+		s.Value = str
+		return nil
+	}
+
+	var content ChatCompletionAssistantMessageParamContent
+	err = json.Unmarshal(data, &content)
+	if err == nil {
+		s.Value = content
+		return nil
+	}
+
+	return fmt.Errorf("cannot unmarshal JSON data as string or assistant content parts")
```

Comment:
if there's no format then let's use errors.New
```suggestion
	return errors.New("cannot unmarshal JSON data as string or assistant content parts")
```